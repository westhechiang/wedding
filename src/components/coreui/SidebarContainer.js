import styled from 'styled-components';
import { theme } from '../../ui';
import sidebarBgLight from '../../assets/images/background/sidebar-bg-light.png';

const SidebarContainer = styled.div`
  width: ${props => (props.width || 'auto')};
  box-sizing: border-box;
  max-height: 100vh;
  overflow-y: ${props => props.overflowY || 'auto'};
  overflow-x: hidden;
  position: relative;
  display: ${props => props.display || 'block'};
  height: ${props => (props.height || 'auto')};
  background-image: url(${sidebarBgLight});
  background-repeat: repeat;
  background-size: 75px;
  box-shadow: ${theme.shadow};
`;

export default SidebarContainer;
