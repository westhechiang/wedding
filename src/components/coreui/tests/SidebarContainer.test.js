import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SidebarContainer from '../../../components/coreui/SidebarContainer';

describe('<SidebarContainer />', () => {
  it('should match its empty snapshot', () => {
    const wrapper = shallow(<SidebarContainer />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its snapshot when passed a "width" prop', () => {
    const wrapper = shallow(<SidebarContainer width="50px" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its snapshot when passed a "height" prop', () => {
    const wrapper = shallow(<SidebarContainer height="50px" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
