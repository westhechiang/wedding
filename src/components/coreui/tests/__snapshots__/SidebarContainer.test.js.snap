// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarContainer /> should match its empty snapshot 1`] = `
<div
  className="sc-bdVaJa jEVwQu"
/>
`;

exports[`<SidebarContainer /> should match its snapshot when passed a "height" prop 1`] = `
<div
  className="sc-bdVaJa dCWhYd"
  height="50px"
/>
`;

exports[`<SidebarContainer /> should match its snapshot when passed a "width" prop 1`] = `
<div
  className="sc-bdVaJa cvQtZk"
  width="50px"
/>
`;
